helper-scripts/deploy_backend.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+# Check if stack name is provided
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <stack-name>"
+    exit 1
+fi
+
+# Check correct AWS account is active
+if [ "$(aws sts get-caller-identity --output text --query 'Account')" != "211125300205" ]; then
+    echo "This script is attempting to be run in a NON DEV account. Please verify your AWS environment configuration"
+    exit 1
+fi
+
+# Ask the user if they want to build and deploy the custom backend-api
+echo
+read -r -p "You are about to deploy a custom Backend API, do you wish to continue? [y]: " yn
+
+if [ "$yn" != "y" ] && [ "$yn" != "Y" ]; then
+    echo "Aborting."
+    exit 1
+fi
+
+STACK_NAME=$1
+
+# Define stack names
+BACKEND_STACK_NAME="${STACK_NAME}-async-backend"
+STS_MOCK_STACK_NAME="${STACK_NAME}-sts-mock"
+
+# Define parameter names
+DEV_OVERRIDE_STS_BASE_URL="DevOverrideStsBaseUrl"
+DEV_OVERRIDE_ASYNC_BACKEND_BASE_URL="DevOverrideAsyncBackendBaseUrl"
+
+# Start deploying backend-api
+echo "Building and deploying custom Backend API stack: $BACKEND_STACK_NAME"
+echo
+cd ../backend-api || exit 1
+sam build --cached
+sam deploy \
+    --stack-name "$BACKEND_STACK_NAME" \
+    --parameter-overrides "$DEV_OVERRIDE_STS_BASE_URL=https://${STS_MOCK_STACK_NAME}.review-b-async.dev.account.gov.uk" \
+    --capabilities CAPABILITY_NAMED_IAM \
+    --resolve-s3
+
+# Ask the user about deploying sts-mock
+deploy_sts_mock=false
+while true; do
```

Comment:
We could simplify this to be an if condition 🤷

Reply:
The reason I used a while loop was to keep re asking the question if the user responds with invalid response. Can you do that with an if?

Reply:
ah fair